### PR TITLE
Merry branch

### PR DESCRIPTION
**Pull request by merryshalabi**
This pull request implements the login, signup, and logout functionality on both the server and client sides. The feature allows users to create an account, log in, and log out seamlessly. It also ensures that access to the home page and recipes is restricted to authenticated users

### CODE REVIEW

**Review by LeoniyaKh — APPROVED** [review]

File: server/models/UserModel.js
Comment (noaliony) [inline server/models/UserModel.js:1]: Please move those changes to user controller :)

**Review by noaliony — APPROVED** [review]

**Review by MikaHurvits — APPROVED** [review]
Amazing Job!

File: server/controllers/rubberDuckController.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
+//const RubberDuck = require('../models/RubberDuckModel');
+=======
 // const RubberDuck = require('../models/RubberDuckModel');
+>>>>>>> origin/main
```
Comment (Noa63) [inline server/controllers/rubberDuckController.js:5]: Think we can delete all the rubber ducks things already

File: client/src/pages/HomePage/HomePage.js [hunk 2]
```diff
@@ -2,24 +2,31 @@ import React, { useEffect, useState } from 'react';
...
+        });
         setRecipes(response.data.recipes);
       } catch (err) {
         setError('Failed to fetch recipes');
         console.error(err);
       }
     };
-    fetchRecipes();
-  }, []);
+    if(user){
```
Comment (Noa63) [inline client/src/pages/HomePage/HomePage.js:25]: Here we said that we're gonna display the recipes no matter if there is a user logged in or not, no?

File: client/src/styles/App.module.css [hunk 3]
```diff
@@ -73,4 +73,208 @@
...
+
+nav a {
+  margin-left: 10px;
+}
+
+
+/* App.module.css */
+
+/* Make sure the header takes the full width of the page */
+.appHeader {
```
Comment (Noa63) [inline client/src/styles/App.module.css:218]: you added some styles to classes that already exist in the top of this file (for example appHeader in line 5, so maybe combine them

**Pull request closed without merging**